packages/components/src/checkbox-control/index.js

```diff
@@ -31,6 +33,26 @@ export default function CheckboxControl( {
 		} );
 	}
 
+	const [ showCheckedIcon, setShowCheckedIcon ] = useState( false );
+	const [ showIndeterminateIcon, setShowIndeterminateIcon ] = useState(
+		false
+	);
+
+	// Run the following callback everytime the `ref` (and the additional
+	// dependencies) change.
+	const ref = useRefEffect(
+		( node ) => {
+			if ( ! node ) {
+				return;
+			}
+
+			node.indeterminate = !! indeterminate;
```

Comment:
It's probably a good idea to leave a comment here about the need to set `indeterminate` as a JS property (and not an HTML attribute)